create_fluent_bit_config.py

```diff
@@ -0,0 +1,273 @@
+        fields = set_fields.split(',')
+        config += """
+[FILTER]
+    Name modify
+    Match *
+"""
+        for field in fields:
+            key, value = field.split(':')
+            config += f"    Set {key.strip()} {value.strip()}\n"
+
+    # Add the OUTPUT section to send logs to Logz.io
+    logzio_logs_token = os.getenv('LOGZIO_LOGS_TOKEN', 'your_logzio_logs_token')
+    logzio_url = os.getenv('LOGZIO_URL', 'https://listener.logz.io:8071')
+    output_id = os.getenv('OUTPUT_ID', 'output_id')
+    headers = os.getenv('HEADERS', '')
+
+    config += f"""
+[OUTPUT]
+    Name  logzio
+    Match *
+    logzio_token {logzio_logs_token}
+    logzio_url   {logzio_url}
+    id {output_id}
+    headers user-agent:logzio-docker-fluentbit-logs
+"""
+
+    # Add custom headers if specified
+    if headers:
+        config += f"    headers      {headers}\n"
+
+    return config
+
+
+# Function to create the multiline parser configuration
+def create_multiline_parser_config():
+    # Base multiline parser configuration
+    multiline_config = """
+[MULTILINE_PARSER]
+    name          multiline-regex
+    type          regex
+    flush_timeout 1000
+    #
+    # Regex rules for multiline parsing
+    # ---------------------------------
+    #
+    # configuration hints:
+    #
+    #  - first state always has the name: start_state
+    #  - every field in the rule must be inside double quotes
+    #
+    # rules |   state name  | regex pattern                  | next state
+    # ------|---------------|--------------------------------------------
+"""
+
+    # Get start state and custom rules from environment variables
+    start_state_rule = os.getenv('MULTILINE_START_STATE_RULE', '([a-zA-Z]+ \\d+ \\d+\\:\\d+\\:\\d+)(.*)')
+    custom_rules = os.getenv('MULTILINE_CUSTOM_RULES', '')
+
+    # Add custom rules as cont state
+    if custom_rules:
+        multiline_config += f'    rule      "start_state"   "/{start_state_rule}/"  "cont"\n'
+        rules = custom_rules.split(';')
+        for rule in rules:
+            multiline_config += f'    rule      "cont"          "{rule.strip()}"                     "cont"\n'
+
+    # If no custom rules, add default start state rule without cont
+    if not custom_rules:
+        multiline_config += f'    rule      "start_state"   "/{start_state_rule}/"\n'
+
+    return multiline_config
+
+
+# Generalized function to save configuration files
+def save_config_file(config, filename):
+    os.makedirs(os.path.dirname(filename), exist_ok=True)
+    with open(filename, 'w') as file:
+        file.write(config)
+    print(f"Configuration file '{filename}' created successfully.")
+
+
+def main():
+    # Generate and save Fluent Bit configuration
+    fluent_bit_config = create_fluent_bit_config()
+    save_config_file(fluent_bit_config, "/fluent-bit/etc/fluent-bit.conf")
+
+    # Generate and save multiline parser configuration if rules are defined
+    if os.getenv('MULTILINE_START_STATE_RULE'):
```

Comment:
I noticed that there are multiple places in the code where environment variables are read using `os.getenv()` and this one specifically as well.
I think it will be cleaner and more efficient to do it once at the beginning of the code, store them in some config object and read the values from there

Reply:
I added configuration object to store environment variables at the beginning of 'create_fluent_bit_config.py'